docs/data/data-grid/editing/editing.md

```diff
@@ -157,18 +157,61 @@ The object also contains a `reason` param that specifies which type of interacti
 The following demo shows how to prevent the user from exiting edit mode when clicking outside of a cell.
 To do this, the `onCellEditStop` prop is used to check if the `reason` is `'cellFocusOut'`.
 If that condition is true, it [disables](/components/data-grid/events/#disabling-the-default-behavior) the default event behavior.
-In this scenario, the user can only stop editing a cell by pressing <kbd class="key">Enter</kbd>, <kbd class="key">Escape</kbd> or <kbd class="key">Tab</kbd>.
+In this context, the user can only stop editing a cell by pressing <kbd class="key">Enter</kbd>, <kbd class="key">Escape</kbd> or <kbd class="key">Tab</kbd>.
 
 {{"demo": "DisableStopEditModeOnFocusOut.js", "bg": "inline"}}
 
-### Disabling default start and stop behavior
+## Controlled mode
 
-The following demo shows how external buttons can be used to start and stop edit mode.
-To do this, the default behavior of the `onCellEditXXX` events is disabled.
-To edit a cell, click on it, then click **Edit**.
+Each cell and row has two modes: `edit` and `view`.
+You can control the active mode using the props `cellModesModel` and `rowModesModel` (only works if `editMode="row"`).
+
+The `cellModesModel` prop accepts an object where the key corresponds to the row ID containing the cell whose mode will be changed.
+Since cell editing works on cells, the value for the key is another object where the key is the column field of the cell.
+Finally, the value for the column field key is another object containing `mode` and additional options.
+The options accepted are the same available in `apiRef.current.startCellEditMode` and `apiRef.current.stopCellEditMode`.
+
+```tsx
+// Changes the mode of field=name from row with id=1 to "edit"
+<DataGrid
+  cellModesModel={{ 1: { name: { mode: GridCellModes.Edit } } }}
```

Comment:
It doesn't look good because of https://github.com/PrismJS/prism/issues/1548

Reply:
Agree!